src/design/place.jl

```diff
@@ -0,0 +1,146 @@
+"""
+    place(A, B, p[; rtol = 1e-3, α::Real = 0, iter::Integer=20 ]) -> K
+
+Places the closed-loop poles of the single- or multi-input system
+
+x = Ax + Bu
+
+at the poles of the vector `p` of desired self-conjugate closed-loop pole
+locations. Computes a gain matrix `F` such that the state feedback `u = –Kx`
+places the closed-loop poles at the locations `p`. In other words, the
+eigenvalues of `A – BK` match the entries of `p` (up to the ordering).
+Place works for multi-input systems and is based on the algorithm from [1].
+
+The non-uniqueness of solutions is used to minimize the sensitivity of
+closed-loop eigenvalues and the norm of the corresponding state feedback matrix
+`F`. The parameter `α ∈ [0,1]` determines the relative weight of these
+criteria. `α=0` only minimizes the norm of `F` and `α=1` only minimizes the
+sensitivity of the closed loop eigenvalues measured by the condition number of
+the eigenvectors of the closed loop eigenvalues.
+
+You can also use place for estimator gain selection by transposing the A matrix
+and substituting C' for B.
+
+l = place(A',C',p).'
+
+# Examples
+```julia
+julia> A = [0 0  0   0   ;
+            1 10 100 1000;
+            0 1  10  100 ;
+            0 0  1   10  ];
+
+julia> B = [1 0;
+            0 1;
+            0 0;
+            0 0];
+
+julia> p = [-1, -2, -1+1im, -1-1im];
+
+julia> F = place(A, B, p);
+
+julia> λ, V = eig(A+B*F);
+
+julia> λ
+Complex{Float64}[4]
+  -2.00 + 0.00im
+  -1.00… + 1.00…im
+  -1.00… - 1.00…im
+  -1.00… + 0.00im
+```
+
+# References
+
+-  [1]: Varga, . "Robust pole assignment via Sylvester equation based state
+        feedback parametrization." IEEE International Symposium on
+        Computer-Aided Control System Design, 2000.
+"""
+function place{M1<:AbstractMatrix,M2<:AbstractMatrix,M3<:AbstractVector}(
+    A::M1, B::M2, p::M3; α::Real = 0.99, rtol::Real=1e-3, iter::Integer=200)
+
+  # /TODO check (A,B) controllable
```

Comment:
here, we can use this `rosenbrock`, etc to get the controllable subspace, etc. I can help with that!